### PR TITLE
Add new tail sampling processor policy: latency

### PR DESCRIPTION
**Pull request by yvrhdn**
**Description:** <Describe what has changed.>

This adds a new policy to the tail sampling processor sampling traces based upon the duration of a span. Adding this policy allows you to sample all traces of which the duration exceeds a given threshold.

It's my first time contributing to Open-Telemetry, let me know if this fits with the other code 😊 

~~**Link to tracking Issue:** <Issue number if applicable>~~

**Testing:** <Describe what testing was performed and which tests were added.>

Added unit tests similar to the other policies.

**Documentation:** <Describe the documentation added.>

I have updated the README.

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
I like this, and the code looks sane. But allow me to make one suggestion: how about taking the *trace* duration, instead of the span? Or at least, have an option for users to make the decision based on the trace or on spans.

**Comment by jpkrohling** [comment]
Alright, let me know when you need a new review. I'm out next week, though, so you might need to ask someone else to review this if you can't wait for me to return.

**Comment by kvrhdn** [comment]
Alright, I have switched the implementation to check the entire duration of a trace :+1: I think this use case will be way more common.

I wasn't sure how to calculate the total duration of a trace, I've implemented it by searching the min/max timestamps of all spans. Does this seem alright?

File: processor/tailsamplingprocessor/README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ The following configuration options are required:
 
 Multiple policies exist today and it is straight forward to add more. These include:
 - `always_sample`: Sample all traces
+- `latency`: Sample based on duration
```
Comment (jpkrohling) [inline processor/tailsamplingprocessor/README.md:16]:
```suggestion
- `latency`: Sample based on the duration of the trace. The duration is determined by looking at the earliest start time and latest end time, without taking into consideration what happened in between.
```

File: processor/tailsamplingprocessor/sampling/latency.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+type latency struct {
+	logger      *zap.Logger
+	thresholdMs int64
+}
+
+var _ PolicyEvaluator = (*latency)(nil)
+
+// NewLatency creates a policy evaluator the samples all traces.
```
Comment (jpkrohling) [inline processor/tailsamplingprocessor/sampling/latency.go:29]:
```suggestion
// NewLatency creates a policy evaluator sampling traces with a duration higher than a configured threshold
```

**Review by jpkrohling — APPROVED** [review]
Looks good! There's a whole world of possible algorithms to determine the real duration of the trace, but this simple "wall clock" approach is correct most of the time for most of the users.

I would just make sure to document this decision in the readme, so that users understand what "trace duration" means for our purposes.

**Comment by kvrhdn** [comment]
> I would just make sure to document this decision in the readme, so that users understand what "trace duration" means for our purposes.

Thanks! Your suggestions have been added 🙂

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**